source/ingress.go

```diff
@@ -249,7 +272,7 @@ func endpointsFromIngress(ing *networkv1.Ingress, ignoreHostnameAnnotation bool,
 			if rule.Host == "" {
 				continue
 			}
-			definedHostsEndpoints = append(definedHostsEndpoints, endpointsForHostname(rule.Host, targets, ttl, providerSpecific, setIdentifier)...)
+			definedHostsEndpoints = append(definedHostsEndpoints, endpointsForIngressHostame(rule.Host, targets, ttl, providerSpecific, setIdentifier)...)
```

Comment:
Why is this logic for keeping targetless endpoints only applied for ingress rules? What about ingress tls spec or ingress hostname annotations?

Reply:
We could only detect this issue in this specific case. The rest of the cases either obtain the information from a different source or have a default behavior that does not end up in deleting the records.

Reply:
This doesn't seem correct.

If I use the hostname annotation, and ignore ingress rules, it uses the same empty target list, and passes to the common `source.endpointsForHostname` lookup here https://github.com/Jimdo/external-dns/blob/master/source/ingress.go#L301

This will return an empty list of endpoints (because no targets) and that will result in the deletion of that DNS record because the endpoint that was there prior is now gone (the thing you are trying to prevent).

Seems like now that you have a global flag for this functionality of handling empty targets, you should just be managing this in that `source.endpointsForHostname` function that these all call, or just in the plan where changes to be made are determined.

Reply:
Ohh, I see. So moving the empty endpoint logic into `source.endpointsForHostname` should work for every case, right?